[Lu Yiting] iP

LuYiting0913 opened the pull request:
### DukePro

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))


DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- FAST SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html)
3. double-click it.
4. add your tasks.
5. let it manage your tasks for you 😉
6. And it is **FREE**!

Features:

- [x]  Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the [main](url) method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

LuYiting0913 committed:
changed gitignore;

LuYiting0913 committed:
Implemented Level-1

LuYiting0913 committed:
Implemented Level-2

LuYiting0913 committed:
Implemented Level-3

LuYiting0913 committed:
Implemented Level-4;

LuYiting0913 committed:
Implemented A-TextUiTesting;

LuYiting0913 committed:
Implemented invalid type exception and empty description exception;

LuYiting0913 committed:
Implemented Level-6;

LuYiting0913 committed:
Implemented IndexOutOfRange error handling;

LuYiting0913 committed:
Implement Level-7. Save

LuYiting0913 committed:
Add data file

LuYiting0913 committed:
Implement Level-8. Date and Times

LuYiting0913 committed:
Merge branch 'branch-Level-7'
Merge Level-7 (Save) commits to the main branch

LuYiting0913 committed:
Fix merge conflict

LuYiting0913 committed:
Merge gradle branch

LuYiting0913 committed:
Implement A-MoreOOP

LuYiting0913 committed:
Add package

LuYiting0913 committed:
Implement A-JUnit

LuYiting0913 committed:
Add java docs

LuYiting0913 committed:
Add more JavaDoc

LuYiting0913 committed:
Tweak code to meet the coding standard

LuYiting0913 committed:
Implement Level-9

LuYiting0913 committed:
Merge branch 'branch-A-CodingStandard'
Merge branch A-CodingStandard to master

LuYiting0913 committed:
Merge branch 'branch-Level-9'
Merge branch Level-9

LuYiting0913 committed:
Add Jar file

LuYiting0913 committed:
Fix bug in file reader class

LuYiting0913 committed:
Add JavaDocs for more methods

LuYiting0913 committed:
Add checkStyle and update style.

cheeheng commented on config/checkstyle.xml:
```diff
@@ -0,0 +1,434 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE module PUBLIC
```
I like your use of checkstyle to ensure coding standards adherence.

cheeheng commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,625 @@
+
+
+/**
+ *
```
Could you add a description of Duke in the JavaDoc comments?

cheeheng commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,625 @@
+ *
+ */
+public class Duke {
+    private ArrayList<Task> all = new ArrayList<>();
```
Could the variable name of this `ArrayList` be changed to `allTasks`? This would make the code more readable. I found several other uses of the variable name `all` in this file.

cheeheng commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,625 @@
+         * @throws FileNotFoundException
+         */
+        public void loadData() throws FileNotFoundException {
+            File f = new File(filePath);
```
Perhaps you could consider using more descriptive names rather than short form names? I found a number of such instances in the code.

cheeheng commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,625 @@
+    }
+
+    private class MarkCommand extends Command {
+        private boolean isMark;
```
I like how this boolean variable is named to sound like a boolean.

cheeheng reviewed (commented):
Overall, this looks good to me. However, there are a few cases where the variable names are short and in contracted form and it does not look obvious to me what they represent. There is also an isolated case where the JavaDoc comment is empty. Consider using more readable variable names if that makes sense.

LuYiting0913 committed:
Implement GUI with JavaFX

LuYiting0913 committed:
Improve GUI

LuYiting0913 committed:
Create new jar file for DUke class.

The previous jar file does not include JavaFX.

Build the jar file using gradle

pratham31012002 commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,780 @@
+
+    private String getResponse(String input) {
+        boolean isExit = false;
+//        try {
+//            String fullCommand = input;
+//            Command c = parser.parse(fullCommand);
+//            c.execute();
+//            isExit = c.isExit();
+//            storage.saveData();
+//        } catch (Exception e) {
+//            ui.showError(e.getMessage());
+//        } finally {
+//        }
```
Perhaps you can decide whether to keep or get rid of this block of code?

pratham31012002 commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,780 @@
+    }
+
+    private class TaskList {
+        private ArrayList<Task> all;
```
Perhaps you can use a better name for this field?

pratham31012002 commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,780 @@
+                fw.write(s);
+                fw.close();
+            } catch (IOException e) {
+                System.out.println("Something went wrong: " + e.getMessage());
```
Perhaps you can handle all print commands in the Ui class?

pratham31012002 commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,780 @@
+         * Execute the add command. Determine the type of task added then add it to taskList.
+         */
+        @Override
+        public void execute() {
+            if (type == TaskType.TODO) {
+                taskList.addTask(new Todo(description));
+            } else if (type == TaskType.DEADLINE) {
+                try {
+                    taskList.addTask(new Deadline(parser.extractDescription(description),
+                                                  parser.extractDate(description)));
+                } catch (DukeException e) {
+                    ui.showError("format incorrect!");
+                }
+            } else if (type == TaskType.EVENT) {
+                try {
+                    taskList.addTask(new Event(parser.extractDescription(description),
+                            parser.extractDate(description)));
+                } catch (DukeException e) {
+                    ui.showError("format incorrect!");
+                }
+            } else {
+                ui.showError("type not supported");
+            }
+        }
```
Perhaps use of switch statements instead of if-else statements can make this code look a lot cleaner?

pratham31012002 commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,780 @@
+            taskList.displayList();
+        }
+
+        @Override
+        public boolean isExit() {
+            return false;
+        }
```
Perhaps consider implementing this in the parent class and only override in `ExitCommand`?

pratham31012002 commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,780 @@
+    private class Parser {
+        private String list, todo, ddl, event, mark, unmark, exit, delete, find;
+
+        public Parser() {
+            list = "list";
+            todo = "todo";
+            event = "event";
+            ddl = "deadline";
+            unmark = "unmark";
+            mark = "mark";
+            exit = "bye";
+            delete = "delete";
+            find = "find";
+        }
```
Perhaps consider using enums instead?

pratham31012002 commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,780 @@
+            return parseStringToDate(s.substring(pos + 4));
+        }
+
+        public String extractDescription(String s) throws DukeException {
+            int pos = s.indexOf("/by");
+            if (pos == -1) {
+                pos = s.indexOf("/at");
+            }
+            if (pos == -1) {
+                throw new DukeException("Invalid structure, include /by or /at before the date");
+            }
+            return s.substring(0, pos - 1);
+        }
+
+        public LocalDate parseStringToDate(String s) {
+            return LocalDate.parse(s.replace('/', '-'));
+        }
+
+        public String parseDateToString(LocalDate date) {
+            return date.format(DateTimeFormatter.ofPattern("MMM d yyyy", Locale.ENGLISH));
+
+        }
```
I like how you have parsed all parts of the input in the Parser class.

pratham31012002 reviewed (commented):
Overall, Good use of OOP principles. There are just a few nits and the code can be made cleaner in few places.

LuYiting0913 committed:
Add assertion in Duke class

This assertion will check whether isExit is false at the end of the run
function.

LuYiting0913 committed:
Improve code quality.

There are repeated code in Parser class. Refactor the variables and
improve code quality.

LuYiting0913 committed:
Merge pull request #1 from LuYiting0913/branch-A-CodeQuality

Improve code quality.

LuYiting0913 committed:
Merge pull request #2 from LuYiting0913/branch-A-Assertions

Add assertion in Duke class

LuYiting0913 committed:
Implement C-Sort.

Sort all tasks based on type and date.

LuYiting0913 committed:
Add more error handling

LuYiting0913 committed:
Change font size and color

LuYiting0913 committed:
Create user guide

LuYiting0913 committed:
Update user guide

LuYiting0913 committed:
Update user guide

LuYiting0913 committed:
Add Ui.png

LuYiting0913 committed:
Modify Ui.png

LuYiting0913 committed:
Add header comments to classes and methods
